### ISSUE TITLE
gateway: IngressServer panics when receiving probe packet on data port

### ISSUE THREAD

**Issue by fstreun** [scionproto/scion#4196]
When SIG A sends a probe packet to the data port of SIG B, SIG B panics with the error message `frame too short`.

This happens when the AS of SIG B is misconfigured and uses the same address for the SIG's `data_addr` and `probe_addr` fields.

topology file:
```
...
  "sigs": {
    "sig1-ff00_0_111": {
      "ctrl_addr": "127.0.0.18:30256",
      "data_addr": "127.0.0.18:30056",
      "probe_addr": "127.0.0.18:30056"
    }
  }
...
```

**Comment by oncilla** [comment]
Hi @fstreun 

Thanks for reporting this.

It is surprising that the gateway would even start in this configuration. Was it able to bind the same port twice?

In any case, this is a DOS vector that a malicious actor could abuse. We should not shutdown the gateway if we receive malformed packets. Rather, we should ignore them.

Care to send a fix?

**Comment by fstreun** [comment]
I came across the bug while implementing a new feature, that allows for defining the remote SIG's addresses in the configuration instead of using the service discovery.
(https://github.com/netsec-ethz/scion/tree/fstreun/sig_static_addresses)

Without my modification, I was able to reproduce the bug with a misconfigured topology file as described above.
The SIG started without problems likely due to the fact that its configuration file did not include the wrong port.
(Note, I run the tests on the netsec-ethz/scion fork)

I might have some time to fix the bug next week.
